src/COVID19/model.py

```diff
@@ -178,7 +191,54 @@ def _get_base_param_from_enum(param):
             break
     return base_name, enum_val
 
-
+class vaccine_schedule(object):
+    def __init__(
+        self,
+        frac_0_9   = 0,
+        frac_10_19 = 0,
+        frac_20_29 = 0,
+        frac_30_39 = 0,
+        frac_40_49 = 0,
+        frac_50_59 = 0,
+        frac_60_69 = 0,
+        frac_70_79 = 0,
+        frac_80    = 0,
+        vaccine_type    = 0,
+        efficacy        = 1.0,
+        time_to_protect = 15,
+        vaccine_protection_period = 365
+    ):
+        fraction_to_vaccinate = [
+            frac_0_9,   frac_10_19, frac_20_29, frac_30_39, frac_40_49,
+            frac_50_59, frac_60_69, frac_70_79, frac_80,
+        ]
+    
+        self.c_fraction_to_vaccinate = covid19.doubleArray( 9 )
+        for age in range( 9 ) :
+            self.c_fraction_to_vaccinate[ age ] = fraction_to_vaccinate[ age ]
+        
+        self.vaccine_type    = vaccine_type
+        self.efficacy        = efficacy
+        self.time_to_protect = time_to_protect
+        self.vaccine_protection_period = vaccine_protection_period
+        
+        self.c_total_vaccinated = covid19.longArray( 9 )
+        for age in range( 9 ) :
+            self.c_total_vaccinated[ age ] = 0
+            
+    def total_vaccinated (self):    
+        total_vaccinated   = [0,0,0,0,0,0,0,0,0]
+        for age in range( 9 ) :
+            total_vaccinated[ age ] = self.c_total_vaccinated[ age ]
+        return total_vaccinated
+    
+    def fraction_to_vaccinate (self):    
+        fraction_to_vaccinate   = [0,0,0,0,0,0,0,0,0]
+        for age in range( 9 ) :
```

Comment:
its a little bit more verbose but doing:
```
for age in AgeGroupEnum:
  fraction_to_vaccinate[ age.value ] = self.c_fraction_to_vaccinate[age.value]
```
protects against any possible changes to the age stratification as the AgeGroupEnum is used everywhere elas

you can also do 

```
fraction_to_vaccinate = [0]*len(AgeGroupEnum) to get the vector set to the correct length as well
```

Again thoughout use len(AgeGroupEnum) rather than directly setting 9 age bins.

Reply:
Done and updated similar examples.